Automattic/WP-Job-Manager
# Allow limiting the number of file uploads per field

Fixes #516

#### Changes proposed in this Pull Request:

* Allows custom multiple file fields to be added with enforced limits on the number of files that can be uploaded.
* Updates `assets/js/jquery-fileupload/jquery.fileupload.js` and `assets/js/jquery-fileupload/jquery.iframe-transport.js` to their newest versions.

#### To Do
- [x] Enforce limit in backend.
- [x] Check in other browsers (Not tested: IE/Edge)

#### Testing instructions:
Snippet to use/adjust:
```
add_filter( 'submit_job_form_fields', function( $fields ) {
  $fields['job']['job_files'] = [
	'label'              => __( 'Job Files', 'wp-job-manager' ),
	'type'               => 'file',
	'required'           => false,
	'placeholder'        => '',
	'priority'           => 6,
	'ajax'               => false,
	'multiple'           => true,
	'file_limit'	     => 2,
	// 'file_limit_message' => 'You can only upload %d FILES, please and thank you',
  ];
  
  return $fields;
} );
```

* With this snippet, try validation with ajax enabled and disabled. Already uploaded files count toward limit and removing them should allow for more files to be uploaded.

Note: `'multiple' => false` will ignore file limit.
